src/main/java/utils/Parser.java

```diff
@@ -197,37 +200,54 @@ private Command prepareDeleteDishCommand(String[] userInputSplit) {
         return new DeleteDishCommand(indexToRemove);
     }
 
-    private Command prepareViewDishCommand() {
+    private Command prepareViewDishCommand(String userInputNoCommand) {
+        try {
+            if (!userInputNoCommand.isBlank()) {
+                throw new DinerDirectorException(Messages.ERROR_COMMAND_INVALID);
+            }
+        } catch (DinerDirectorException e) {
+            return new IncorrectCommand();
+        }
         return new ViewDishCommand();
     }
 
-    private Command prepareAddDishCommand() {
+    private Command prepareAddDishCommand(String userInputNoCommand) {
         //call for subsequent inputs:
         // name of dish?
         // price of dish?
         // list of ingredients, maybe indicate number of ingredients followed by listing them
 
+        try {
+            if (!userInputNoCommand.isBlank()) {
+                throw new DinerDirectorException(Messages.ERROR_COMMAND_INVALID);
+            }
+        } catch (DinerDirectorException e) {
+            return new IncorrectCommand();
+        }
+
         String name;
         int price;
         ArrayList<String> ingredients;
 
         try {
-            System.out.println("Name of Dish?");
             Scanner userInput = new Scanner(System.in);
+            System.out.println("Name of Dish?");
```

Comment:
maybe you can follow a similar format to our other commands, n/dish_name p/price. then for ingredients you can do what you are doing now